File: leet_code/math/9/9_1.dart
class Solution {
  bool isPalindrome(int x) {
    if (x < 0) {
      return false;
    }
    int original_num = x;
    int reversed_num = 0;
    while (x > 0) {
      reversed_num = reversed_num * 10 + x % 10;
      x ~/ 10;
    }
    return original_num == reversed_num;
  }
}

void main() {
  Solution s = Solution();
  print(s.isPalindrome(121));
}

File: leet_code/math/9/9_1.py
class Solution:
    def isPalindrome(self,x:int)->bool:
        if x<0:
            return False
        original_num = x 
        reversed_num = 0
        while x>0:
            reversed_num = reversed_num*10+x%10
            x //= 10
        return original_num == reversed_num


s = Solution()
print(s.isPalindrome(121))